test/unit/util/test_cuda_vector_matrix.cpp

```diff
@@ -0,0 +1,194 @@
+#include <gtest/gtest.h>
+
+#include <micm/util/cuda_vector_matrix.hpp>
+
+#include "test_matrix_policy.hpp"
+
+template<class T>
+using Group1MatrixAlias = micm::CudaVectorMatrix<T,1>;
+template<class T>
+using Group2MatrixAlias = micm::CudaVectorMatrix<T,2>;
+template<class T>
+using Group3MatrixAlias = micm::CudaVectorMatrix<T,3>;
+template<class T>
+using Group4MatrixAlias = micm::CudaVectorMatrix<T,4>;
+
+
+
+TEST(VectorMatrix, SmallVectorMatrix)
+{
+  auto matrix = testSmallMatrix<Group2MatrixAlias>();
+
+  matrix.CopyToDevice();
+  matrix.GetFromDevice();
```

Comment:
Would it be worth adding a kernel function just for tests that modified the data on the device before copying it back to the host? As these tests are currently set up, `CopyToDevice()` and `CopyFromDevice()` could do nothing and it seems like the tests would pass.

Reply:
I was thinking about that as well and am definitely interested in adding that functionality in so we can test this.  I wasn't sure how to best add it without creating a dummy function in the cuda files though.  Not sure the current `cuda_vector_matrix` class is the right place but more than open to ideas on how to best go about it.

Reply:
Maybe adding it to the test folder? something like:
```
test/unit/util/cuda/cuda_matrix_utils.cu
test/unit/util/cuda/cuda_matrix_utils.cuh
```
and you could just create a wrapper function that takes a `CudaMatrixParam&` and passes the device pointer (which points to an array of doubles) to a kernel function that then adds some specific values to certain elements? (I think it's ok for this function to assume some size for the array, since it's just for the tests).

Reply:
Added in the test that we discussed!